pkg/speaker/subnet.go

```diff
@@ -348,11 +348,18 @@ func getNextHopFromPathAttributes(attrs []bgp.PathAttributeInterface) net.IP {
 	return nil
 }
 
-func getNextHopAttribute(neighborAddress, routeID string) string {
-	nextHop := routeID
+func (c *Controller) getNextHopAttribute(neighborAddress string) string {
+	nextHop := c.config.RouterID
 	routes, err := netlink.RouteGet(net.ParseIP(neighborAddress))
 	if err == nil && len(routes) == 1 && routes[0].Src != nil {
 		nextHop = routes[0].Src.String()
 	}
+	proto := util.CheckProtocol(nextHop)
+	nextHopIP := net.ParseIP(nextHop)
+	nodeIP := c.config.NodeIPs[proto]
+	podIP := c.config.PodIPs[proto]
+	if nodeIP != nil && podIP != nil && nextHopIP != nil && nextHopIP.Equal(podIP) {
+		nextHop = nodeIP.String()
+	}
```

Comment:
I think we should use nodeIP as nextHtop if the nodeIP is not nil. No need to check podIP.

Reply:
done